### PR TITLE
Fixes Reagent-Filled cells always exploding even without plasma.

### PR DESCRIPTION
**Pull request by pitreforten**
## About The Pull Request

This PR fixes reagent-filled cells always exploding, even without plasma.

## Why It's Good For The Game

It fixes a bug.

## Changelog
:cl: Hannuwu
fix: Reagent-Filled cells always exploding, even without plasma.
/:cl:

### CODE REVIEW

File: code/modules/power/cell.dm [hunk 1]
```diff
@@ -103,10 +103,10 @@
 	return (FIRELOSS)
 
 /obj/item/stock_parts/cell/on_reagent_change(changetype)
-	rigged = !isnull(reagents.has_reagent(/datum/reagent/toxin/plasma, 5)) //has_reagent returns the reagent datum
+	if (reagents.has_reagent(/datum/reagent/toxin/plasma, 5))
+		rigged = TRUE //has_reagent returns 0 (FALSE) if there is no reagent
```
Comment (YPOQ) [inline code/modules/power/cell.dm:107]:
```suggestion
		rigged = TRUE //has_reagent returns 0 (FALSE) if there is no reagent
	else
		rigged = FALSE
```
to handle the case of removing plasma with a syringe (this is possible, right?)

Comment (ATH1909) [inline code/modules/power/cell.dm:107]:
```suggestion
	rigged = reagents.has_reagent(/datum/reagent/toxin/plasma, 5))

```
might wanna make sure that you can't use this to un-rig cells that have been rigged by means other than a plasma injection, though

**Review by Cyberboss — CHANGES_REQUESTED** [review]
@ATH1909

File: code/modules/power/cell.dm (hunk 1 above)
Comment (ShizCalev) [reply code/modules/power/cell.dm:107]: if(!rigged)
    rigged = reagents.has_reagent(/datum/reagent/toxin/plasma, 5)

File: code/modules/power/cell.dm [hunk 2]
```diff
@@ -103,9 +103,9 @@
 	return (FIRELOSS)
 
 /obj/item/stock_parts/cell/on_reagent_change(changetype)
-	rigged = !isnull(reagents.has_reagent(/datum/reagent/toxin/plasma, 5)) //has_reagent returns the reagent datum
-	..()
+	rigged = reagents.has_reagent(/datum/reagent/toxin/plasma, 5))
 
+	..()
```
Comment (ShizCalev) [inline code/modules/power/cell.dm:108]:
```suggestion
	return ..()

```

File: code/modules/power/cell.dm [hunk 3]
```diff
@@ -103,9 +103,9 @@
 	return (FIRELOSS)
 
 /obj/item/stock_parts/cell/on_reagent_change(changetype)
-	rigged = !isnull(reagents.has_reagent(/datum/reagent/toxin/plasma, 5)) //has_reagent returns the reagent datum
-	..()
+	rigged = reagents.has_reagent(/datum/reagent/toxin/plasma, 5))
```
Comment (ShizCalev) [inline code/modules/power/cell.dm:106]: rigged can be set by corrupt(), add handling for that so we don't override it as well.

Comment (ShizCalev) [inline code/modules/power/cell.dm:106]: should add a var to handle if rigged was set by corrupt() so we don't end up overriding that if the chems were removed

**Review by ShizCalev — CHANGES_REQUESTED** [review]

**Pull request closed without merging**